test/test_listener_deci2.cpp

```diff
@@ -58,16 +58,16 @@ TEST(Listener, DeciThenListener) {
     EXPECT_FALSE(s.check_for_listener());
     EXPECT_TRUE(l.connect_to_target());
     // kind of a hack.
-    while(!s.check_for_listener()) {
-      printf("...\n");
+    while (!s.check_for_listener()) {
+      // printf("...\n");
```

Comment:
This isn't an issue for me (it prints only zero time or once), but I imagine the networking stack on the github actions machines might be a lot busier than my desktop.  The test on github actions takes 1.5 seconds, compared to 1 second on my computer, so I don't think there's a serious delay here.  Just getting rid of the silly print seems like a good plan.